### PR TITLE
Update repo to ODK 1.5 release

### PR DESCRIPTION
**Pull request by matentzn**
Updates ODK to last weeks major ODK release 1.5. Currently the production release `latest` is en par with the `dev` snapshot. After considering shifting to `production` in #465, I know believe this is just churn as we will probably need to stay on the edge of ODK as time progresses (1.5 was the first major release in 1 year).

## Overview

This PR:
- Updates the remaining files to ODK 1.5 (development snapshot, currently `en par` with production).

### Documentation
Was the documentation added/updated under `docs/`?
- [ ] Yes
- [X] No, updates to the docs were not necessary after careful consideration

### QC
Was the full pipeline run before submitting this PR using `sh run.sh make build-mondo-ingest` on this branch (after 
`docker pull obolibrary/odkfull:dev`), and no errors occurred?
- [x] Yes
- [ ] No, there are no functional (code-related) changes to the pipeline in the PR, so no re-run is necessary
   
### New Packages
Were any new *Python packages* added?
- [ ] Yes, [requirements.txt.full](https://github.com/INCATools/ontology-development-kit/blob/master/requirements.txt.full) was updated and an [ODK issue](https://github.com/INCATools/ontology-development-kit/issues/new) submitted
- [X] No

Were any other *non-Python packages* added?
- [ ] Yes, [Dockerfile](https://github.com/INCATools/ontology-development-kit/blob/master/Dockerfile) updated and an [ODK issue](https://github.com/INCATools/ontology-development-kit/issues/new) submitted
- [X] No

### PR Review and Conversations Resolved
Has the PR been sufficiently reviewed by at least 1 team member of the Mondo Technical team and all threads resolved?
- [x] Yes

### CODE REVIEW

**Comment by twhetzel** [comment]
@matentzn FYI- I don't see an Assignee. Do you want this reviewed and merged?

**Comment by matentzn** [comment]
> @matentzn FYI- I don't see an Assignee. Do you want this reviewed and merged?


Sorry, forgot to assign. I assigned myself to the pr, and added you as a reviewer.

File: src/ontology/Makefile [hunk 1]
```diff
@@ -63,7 +63,7 @@ FORMATS_INCL_TSV = $(sort $(FORMATS) tsv)
 RELEASE_ARTEFACTS = $(sort $(ONT)-base $(ONT)-full )
 
 ifeq ($(ODK_DEBUG),yes)
-ODK_DEBUG_FILE = $(TMPDIR)/debug.log
```
Comment (joeflack4) [inline src/ontology/Makefile:66]: This and many other changes are coming from running `update_repo`?

File: src/scripts/match-mondo-sources-all-lexical.py [hunk 2]
```diff
@@ -147,29 +147,5 @@ def run(input: str, config: str, rules: str, rejects: str, output: str):
     with open(str(Path(output.replace("lexical", "lexical-2"))), "w") as f:
         filter_file(input=str(Path(output)), output=f, **kwargs)
 
-
-def iri_to_curie(item):
```
Comment (joeflack4) [inline src/scripts/match-mondo-sources-all-lexical.py:151]: Oh ok good, we're finally getting rid of this. I see it had no usages as well.

File: src/ontology/mondo-ingest.Makefile [hunk 3]
```diff
@@ -91,7 +91,7 @@ $(COMPONENTSDIR)/ordo.owl: $(TMPDIR)/ordo_relevant_signature.txt config/properti
 			--update ../sparql/ordo-construct-subsets.ru \
 			--update ../sparql/ordo-construct-d2g.ru \
 			--update ../sparql/ordo-replace-annotation-based-mappings.ru \
-		filter -T $(TMPDIR)/ordo_relevant_signature.txt --trim false \
+		remove -T $(TMPDIR)/ordo_relevant_signature.txt --select complement --select "classes individuals" --trim false \
```
Comment (joeflack4) [inline src/ontology/mondo-ingest.Makefile:94]: I'm guessing there are a few things in this PR that are not specifically related to the ODK version up? Or maybe this one is related to a `robot` version up.

File: src/ontology/Makefile (hunk 1 above)
Comment (matentzn) [reply src/ontology/Makefile:66]: All changes to Makefile yes!

File: src/scripts/match-mondo-sources-all-lexical.py (hunk 2 above)
Comment (matentzn) [reply src/scripts/match-mondo-sources-all-lexical.py:151]: Yes, that's right.

File: src/ontology/mondo-ingest.Makefile (hunk 3 above)
Comment (matentzn) [reply src/ontology/mondo-ingest.Makefile:94]: This line caused an issue where some important pieces of the module where removed (synonym types), which caused the new version of oak to fail. This is the right way of doing it anyways, other ingests use the same filter step.

File: src/ontology/Makefile [hunk 4]
```diff
@@ -638,7 +645,7 @@ ROBOT_RELEASE_IMPORT_MODE_BASE=$(ROBOT) remove --input $< --select imports --tri
 # base: A version of the ontology that does not include any externally imported axioms.
 $(ONT)-base.owl: $(EDIT_PREPROCESSED) $(OTHER_SRC) $(IMPORT_FILES)
 	$(ROBOT_RELEASE_IMPORT_MODE) \
-	reason --reasoner ELK --equivalent-classes-allowed asserted-only --exclude-tautologies structural \
+	reason --reasoner ELK --equivalent-classes-allowed asserted-only --exclude-tautologies structural --annotate-inferred-axioms False \
```
Comment (twhetzel) [inline src/ontology/Makefile:648]: why is `--annotate-inferred-axioms False` added?

**Review by twhetzel — APPROVED** [review]
lgtm

**Comment by matentzn** [comment]
Full pipeline was run on this branch #468

**Pull request merged**